File: practice/algorithms/implementation/kangaroo/Kangaroo.js
'use strict';

const processData = input => {
    const lines = input.split('\n');
    const [x1, v1, x2, v2] = lines[0].split(' ').map(i => parseInt(i));
    console.log(sameLocation(x1, v1, x2, v2));
};

const sameLocation = (x1, v1, x2, v2) => {
    if(v1 < v2) return "NO";
    if(x1 < x2 && v1 > v2) {
        const xDiff = Math.abs(x1 - x2);
        const vDiff = Math.abs(v1 - v2);
        if(xDiff % vDiff == 0) {
            return "YES";
        } else {
            return "NO";
        }
    }
    return "NO";
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/kangaroo/Kangaroo.java
import java.util.*;

public class Kangaroo {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int x1 = stdin.nextInt();
        int v1 = stdin.nextInt();
        int x2 = stdin.nextInt();
        int v2 = stdin.nextInt();

        System.out.println(isSameLocation(x1, v1, x2, v2));
        stdin.close();
    }

    private static String isSameLocation(int x1, int v1, int x2, int v2) {
        if(v1 < v2) return "NO";
        if(x1 < x2 && v1 > v2) {
            int xDiff = Math.abs(x1 - x2);
            int vDiff = Math.abs(v1 - v2);
            if(xDiff % vDiff == 0) {
                return "YES";
            } else {
                return "NO";
            }
        }
        return "NO";
    }
}